### PR TITLE
Automated Package Publishing

### LINKED ISSUE #10331: Automate NPM Publishing of SDKs in monorepo
(@0xarthurxyz here: editing the description)

Aspirational goal: automate NPM releases.
Note: This is a large initiative and might be blocked or influenced by decisions to split the monorepo into single repositories.

Automatic NPM publishing for the monorepo is still pending (or the status is currently unknown). 
Currently monorepo SDKs are published manually as before.

Goal:
- [ ] Support all packages (including @celo/abis & @celo/contracts)

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

Github Action to Automatically publish npm packages. Workflow is like so

* contributors add changesets to PRs with `yarn cs add` 

* each time a PR is merged into master the action being added in this pr will Open or update a PR ( aka Version Packages PR) which applies those changeset files to A)  bump versions as specified by changeset files and B) create a changelog. 

* merging the Version Packages  will result in all packages that require publishing being published, and a github releases being created

### Tested

used on social connect repo 

### Related issues

- Fixes #10331

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -10,17 +10,17 @@ This will bring up an interactive console which asks which packages are affect a
 
 ## Auto Releasing
 
-(coming soon)
+The Release.yaml workflow will create a PR called "Version Packages", each time a PR is merged to master with changeset files this PR will be rebased and updated to show what the versions would be of published then. Merging this PR in will lead to packages being built and published to npm and github release notes being published.
```
Comment (nicolasbrugneaux) [inline RELEASE.md:13]: woop

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ env.NPM_TOKEN }}
+        with:
+          # This expects you to have a script called release which does a build for your packages and calls changeset publish
+          publish: yarn release
+      - name: Remove Yarn
+        # https://github.com/actions/setup-node/issues/182
+        if: always()
+        run: npm uninstall -g yarn
```
Comment (jcortejoso) [inline .github/workflows/release.yaml]: The runner lifecycle is the job lifecycle (created and destroyed for each job), so probably we can skip removing anything?

**Review by jcortejoso — APPROVED** [review]
👍

File: .github/workflows/release.yaml (hunk 2 above)
Comment (aaronmgdr) [reply .github/workflows/release.yaml]: oh According the issue above for self hosted runners it isnt destroyed? 

but i dont know

**Pull request merged**